internal/proto/internal.proto

```diff
@@ -246,3 +248,28 @@ message Rate {
   RateType rt = 1;
   double r = 2;
 }
+
+// clustering info of data distribution
+message ClusteringInfo {
```

Comment:
not very clear between 'ClusteringInfo' and 'ClusteringGroup',  here ClusteringInfo means one clustering operation(which will generate a group of clusters), 'ClusteringGroup' contains multiple 'ClusteringInfo', so we should have a list of groupID? we can have a map<groupId, ClusteringInfo>

Reply:
`ClusteringGroup` is all data generated by one clustering operation, contains multi `ClusteringInfo`. groupID is unique id of `ClusteringGroup`. `ClusteringInfo ` has its own id. So ClusteringInfos in the same ClusteringGroup have different id and share one groupID